src/lib/shared/utils.ts

```diff
@@ -100,15 +100,15 @@ export function base64StringToTransaction<T extends Transaction>(transactionByte
  * */
 export function transactionToTransactionBody<T extends Transaction>(
   transaction: T,
-  nodeAccountId: AccountId,
+  nodeAccountId: AccountId, // To mantain compatibility
 ) {
   // This is a private function, though provides the capabilities to construct a proto.TransactionBody
   //@ts-ignore
-  return transaction._makeTransactionBody(nodeAccountId)
+  return transaction._signedTransactions.current.bodyBytes
```

Comment:
This utility function is used to get the transaction body before the transaction is signed, based on the name of _signedTransactions, I'd imagine this internal function only has the transactions in this array if the transaction has already been signed?

If so, we may need to try another method

@MiguelLZPF

Reply:
Hi @tmctl! if I understood you correctly, there is a problem getting the signed transaction body here because this utility function was intended to get the transaction body not the signed transaction body.
So, the original issue is that HederaWalletConnect is getting the transaction body also when it should be looking for the signed transaction body.
So the solution is to have two separate utility functions then, right?

Reply:
Added in the latest commit, take a look at it and tell me what you guys think 🤔

Reply:
I've reverted the change; the issue was with the naming. `transaction._signedTransactions.current.bodyBytes` actually contains the UNSIGNED transaction body bytes, which makes sense since this occurs before the transaction is signed. It was confusing initially because of the misleading name, as you pointed out in your previous message.

Reply:
This is the signed transaction bytes: `signedTransaction._transactions.current.signedTransactionBytes` 🫣. In my opinion is mega-difficult to understand this structure...